### PR TITLE
BC-7936 Refactor calendar test to handle multiple selectors for form fields.

### PR DESCRIPTION
**Pull request by UzaeirKhan**
# Description

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/BC-7936

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: static/scripts/calendar.js [hunk 1]
```diff
@@ -49,6 +49,25 @@ const getCalendarLanguage = (langAttribute) => {
 	}
 };
 
+// Add data-testid attributes to navigation buttons in calendar page
```
Comment (hoeppner-dataport) [inline static/scripts/calendar.js:52]:
```suggestion
// Add data-testid attributes to navigation buttons in calendar page as the calendar module
// that is used (https://fullcalendar.io) does not support configuration of data-testids for buttons in used version
```

File: static/scripts/calendar.js [hunk 2]
```diff
@@ -49,6 +49,25 @@ const getCalendarLanguage = (langAttribute) => {
...
+		{ selector: '.fc-right .fc-today-button', testId: 'right-today-button' },
+		{ selector: '.fc-right .fc-timeGridDay-button', testId: 'right-day-view-button' },
+		{ selector: '.fc-right .fc-timeGridWeek-button', testId: 'right-week-view-button' },
+		{ selector: '.fc-right .fc-dayGridMonth-button', testId: 'right-month-view-button' },
+	];
+
+	buttonData.forEach(({ selector, testId }) => {
+		const button = document.querySelector(selector);
+		if (button) {
+			button.setAttribute('data-testid', testId);
```
Comment (hoeppner-dataport) [inline static/scripts/calendar.js:66]: Better - but maybe not as readable... just a suggestion.
```suggestion
			button.dataset.testid = testId;
```

File: static/scripts/calendar.js [hunk 3]
```diff
@@ -143,8 +167,12 @@ $(document).ready(() => {
 				return false;
 			}
 			// personal event
-			const startDate = toMoment(event.start, calendar).format($t('format.dateTimeToPicker'));
-			const endDate = toMoment(event.end || event.start, calendar).format($t('format.dateTimeToPicker'));
+			const startDate = toMoment(event.start, calendar).format(
```
Comment (hoeppner-dataport) [inline static/scripts/calendar.js:170]: Check your formater settings: if you do not change anything... the code should not be formated differently

File: static/scripts/calendar.js [hunk 4]
```diff
@@ -189,8 +219,12 @@ $(document).ready(() => {
 			const { date } = info;
 
 			// open create event modal
-			const startDate = toMoment(date, calendar).format($t('format.dateTimeToPicker'));
-			const endDate = toMoment(date, calendar).add(1, 'hour').format($t('format.dateTimeToPicker'));
+			const startDate = toMoment(date, calendar).format(
```
Comment (hoeppner-dataport) [inline static/scripts/calendar.js:222]: same as above...

**Review by hoeppner-dataport — CHANGES_REQUESTED** [review]
Please undo the changes that were caused by incorrect formater.
Besides that -it's only suggestions.

File: static/scripts/calendar.js (hunk 1 above)
Comment (UzaeirKhan) [reply static/scripts/calendar.js]: done

File: static/scripts/calendar.js (hunk 2 above)
Comment (UzaeirKhan) [reply static/scripts/calendar.js]: done

File: static/scripts/calendar.js (hunk 3 above)
Comment (UzaeirKhan) [reply static/scripts/calendar.js]: thanks for pointing it out, corrected my mistake.

File: static/scripts/calendar.js (hunk 4 above)
Comment (UzaeirKhan) [reply static/scripts/calendar.js]: done

File: static/scripts/calendar.js [hunk 5]
```diff
@@ -49,6 +49,26 @@ const getCalendarLanguage = (langAttribute) => {
 	}
 };
 
+// Add data-testid attributes to navigation buttons in calendar page as the calendar module
+// that is used (https://fullcalendar.io) does not support configuration of data-testids for buttons in used version
```
Comment (muratmerdoglu-dp) [inline static/scripts/calendar.js:53]: I think the current version of the package (4.4.2) should be mentioned here.

File: static/scripts/calendar.js [hunk 6]
```diff
@@ -49,6 +49,26 @@ const getCalendarLanguage = (langAttribute) => {
...
+const addDatatestidsToCalendarNavigation = () => {
```
Comment (muratmerdoglu-dp) [inline static/scripts/calendar.js:54]:
```suggestion
const addDataTestIdsToCalendarNavigation = () => {
```
... to better readability.

File: static/scripts/calendar.js [hunk 7]
```diff
@@ -207,6 +227,12 @@ $(document).ready(() => {
 		viewRender(info) {
 			window.location.hash = info.view.name;
 		},
+
+		// viewSkeletonRender to trigger the addition of data-testid to navigation buttons
+		viewSkeletonRender() {
+			// Add custom data-testids for navigation buttons
+			addDatatestidsToCalendarNavigation();
+		},
```
Comment (muratmerdoglu-dp) [inline static/scripts/calendar.js:235]: The comment lines are not necessary. The method name explains its aim itself :)

File: static/scripts/calendar.js [hunk 8]
```diff
@@ -304,4 +330,4 @@ $(document).ready(() => {
 			hideAreaByEvent(event);
 		}
 	});
-});
+});
```
Comment (muratmerdoglu-dp) [inline static/scripts/calendar.js:333]: Please add one empty line at the end of the file to stop the Eslint complaining.

**Review by muratmerdoglu-dp — CHANGES_REQUESTED** [review]

File: static/scripts/calendar.js (hunk 6 above)
Comment (UzaeirKhan) [reply static/scripts/calendar.js:54]: Done

File: static/scripts/calendar.js (hunk 7 above)
Comment (UzaeirKhan) [reply static/scripts/calendar.js]: Done

File: static/scripts/calendar.js (hunk 8 above)
Comment (UzaeirKhan) [reply static/scripts/calendar.js]: Done

**Review by muratmerdoglu-dp — APPROVED** [review]

**Pull request merged**